veloroutes_voies_vertes/install/sql/veloroutes/10_FUNCTION.sql

```diff
@@ -57,6 +57,32 @@ $$;
 COMMENT ON FUNCTION veloroutes.revet() IS 'Force le revêtement à être NULL si le segment est en projet ou fictif';
 
 
+-- v_portion_insert()
+CREATE FUNCTION veloroutes.v_portion_insert() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$DECLARE seg RECORD;
+
+BEGIN
+    -- insert the new portion first
+    INSERT INTO veloroutes.portion(nom, description,type_portion)
+    VALUES(NEW.nom, NEW.description, NEW.type_portion);
+	
+	--cursor for each segment that compose the new entity
+	FOR seg IN( 
+	SELECT veloroutes.segment.id_local
+		FROM veloroutes.segment, veloroutes.v_portion
+		WHERE ST_Within(veloroutes.segment.geom, NEW.geom))
+		
+	LOOP
+	--insert in table element as many rows as there are segments in seg 
+		INSERT INTO veloroutes.element(id_portion,id_segment) VALUES(currval('veloroutes.portion_id_local_seq'),seg.id_local);
+	END LOOP;
+	
+ 	RETURN NEW;
+	
+END;$$;
```

Comment:
```suggestion
    AS $$
    DECLARE pid int;

BEGIN
    -- insert the new portion first
    INSERT INTO veloroutes.portion(nom, description,type_portion)
    VALUES(NEW.nom, NEW.description, NEW.type_portion);
    RETURNING id_local into pid;
    
    INSERT INTO veloroutes.element(id_portion,id_segment) 
    SELECT pid, veloroutes.segment.id_local
		FROM veloroutes.segment s
		WHERE ST_Within(s.geom, NEW.geom);
	
 	RETURN NEW;
	
END;$$;
```
Quelque chose comme ça

Reply:
L'ajout dans élément ne fonctionne toujours pas mais si je change la condition WHERE ça fonctionne.
Pourtant j'ai testé ST_Within dans la console, il doit y avoir y avoir un résultat pour cette requête..